src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,83 @@
+package lotto.controller;
+
+import lotto.domain.BonusNumber;
+import lotto.domain.Lotto;
+import lotto.domain.LottoMachine;
+import lotto.domain.LottoResult;
+import lotto.domain.Lottos;
+import lotto.domain.Money;
+import lotto.parser.InputParser;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+public class LottoController {
+
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final LottoMachine lottoMachine;
+
+    public LottoController(InputView inputView, OutputView outputView, LottoMachine lottoMachine) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.lottoMachine = lottoMachine;
+    }
+
+    public void run() {
+        Money money = inputPurchaseAmount();
+        Lottos lottos = lottoMachine.issue(money);
+        outputView.printLottos(lottos);
+
+        Lotto winningLotto = inputWinningNumbers();
+        BonusNumber bonusNumber = inputBonusNumber(winningLotto);
+
+        LottoResult result = lottos.calculateResult(winningLotto, bonusNumber);
+        printResults(result, money);
+    }
+
+    private <T> T retryUntilValid(Supplier<T> task) {
+        while (true) {
+            try {
+                return task.get();
+            } catch (IllegalArgumentException e) {
+                outputView.printErrorMessage(e.getMessage());
+            }
+        }
+    }
+
+    private Money inputPurchaseAmount() {
+        return retryUntilValid(() -> {
+            String input = inputView.readLottoPurchaseAmount();
+            int amount = InputParser.parseSingleNumber(input);
+
+            return new Money(amount);
+        });
+    }
```

Comment:
```suggestion
    private Money inputPurchaseAmount() {
          String input = inputView.readLottoPurchaseAmount();
          int amount = InputParser.parseSingleNumber(input);

          return new Money(amount);
    }
```
```java
Money money = retryUtillValid(inputPurchaseAmount());
```
이런식으로 바꿔볼 수 있을것 같습니다!
inputWinningNumbers()라는 메서드 이름에 맞게 당첨번호를 입력하는 기능만 구현을하고
실제 재시도하는 로직은 inputwinningNumbers()라는 로직을 사용한다는 개념으로 바꾸면 어떨까 싶습니다.

Reply:
와, 정말 좋은 방법 같네요!
말씀하신 대로 입력 부분만 `retryUntilValid()` 로직에 담는 방식이 훨씬 깔끔하고 책임도 명확해지는 것 같습니다 🙂
덕분에 더 좋은 구조를 배웠습니다 💡